File: backend-php-2/app/Helpers/HandleResponse.php
<?php

namespace App\Helpers;

class HandleResponse
{
    public static function success($traceCode, $data, $message = null, $code = 200)
    {
        return response()->json([
            'error' => false,
            'traceCode' => $traceCode,
            'message' => $message,
            'data' => $data
        ], $code);
    }

    public static function error($traceCode, $message, $code)
    {
        return response()->json([
            'error' => true,
            'traceCode' => $traceCode,
            'message' => $message,
        ], $code);
    }
}

File: backend-java/crud_java/src/main/java/com/example/crud_java/helpers/HandleResponse.java
package com.example.crud_java.helpers;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

// ...
public class HandleResponse {
    public static ResponseEntity<Object> success(String traceCode, Object data, String message, HttpStatus httpStatus) {
        ApiResponse response = new ApiResponse(false, traceCode, message, data);
        return new ResponseEntity<>(response, httpStatus);
    }

    public static ResponseEntity<Object> error(String traceCode, String message, HttpStatus httpStatus) {
        ApiResponse response = new ApiResponse(true, traceCode, message, null);
        return new ResponseEntity<>(response, httpStatus);
    }
}
